Adding feature to allow reports to be emailed

AbhiAgarwal opened the pull request:
On the project I am working on we wanted to add the ability to email the user rather than generate it and get a download (either asynchronously or synchronously). I've taken a first try at implementing this into django-report-builder. I added a `email_report` report to `views.py` that is called if the user sets the setting `REPORT_BUILDER_EMAIL_NOTIFICATION` to `True` in `settings.py`.

By default the email is sent to the current user's `user.email`. I have already defined the subject headers, and the body for the default email that is sent. In addition, I also attempted to add templating, but tried to be as abstract as possible and as less reliant on other templating libraries as possible. 

The templating, as I implemented it, is the user specifying a string with HTML in the settings file: `REPORT_BUILDER_EMAIL_TEMPLATE`. We're going to be implementing it using django-dbtemplates, but I didn't want to force other django projects to use that.

I would love some feedback on the code I've already implemented, and things I should improve on or packages I should utilize. I'll improve them, and would love this to be merged at some point.

bufke commented on docs/quickstart.md:
```diff
@@ -63,6 +63,24 @@ Advantages of this option
+
+    REPORT_BUILDER_EMAIL_NOTIFICATION = True
+    REPORT_BUILDER_EMAIL_SUBJECT = ""
+    REPORT_BUILDER_EMAIL_TEMPLATE = "<p>The report is <a href='{0}'>here</u></p>"
```
Let's move this to a django template instead of right in the settings.

bufke commented on report_builder/tests.py:
```diff
@@ -674,3 +676,27 @@ def test_formatter(self):
+class ViewTests(TestCase):
+    def test_email_report_without_template(self):
+        settings.REPORT_BUILDER_EMAIL_NOTIFICATION = True
+        emailSubject = getattr(settings, 'REPORT_BUILDER_EMAIL_SUBJECT', False) or "Report is ready"
```
No camel case please.

bufke commented:
Will merge once changes are made. Thanks.

AbhiAgarwal commented:
@bufke I have updated it to using django-templates. I've updated the tests to reflect that and added a sample django-template as default. Also simple to overwrite if needed by any django app that needs to.
